src/main/java/com/google/sps/data/TestData.java

```diff
@@ -0,0 +1,74 @@
+    private static final ArrayList<String> TAG_LIST = new ArrayList<String>(
+        Arrays.asList("test", "name", "search"));
+    private static final ArrayList<String> INGRED_LIST = new ArrayList<String>(
+        Arrays.asList("none"));
+    private static final ArrayList<String> STEPS_LIST = new ArrayList<String>(
+        Arrays.asList("none"));
+    private static final String ENTITY_NAME = "PublicRecipe";
+    private static final String RECIPE_NAME = "recipes for testing name search";
+    private static final String ALT_RECIPE_NAME = "alt recipes for testing name search";
+
+    // Recipes which are used for testing in this class
+    private static final Entity RECIPE_001 = TestUtil.createRecipeEntity(
+      ENTITY_NAME, RECIPE_NAME, "THIS IS TEST RECIPE 001 FOR THE SEARCH BY NAME TEST",
+      TAG_LIST, INGRED_LIST, STEPS_LIST, -1);
+    private static final Entity RECIPE_002 = TestUtil.createRecipeEntity(
+      ENTITY_NAME, RECIPE_NAME, "THIS IS TEST RECIPE 002 FOR THE SEARCH BY NAME TEST",
+      TAG_LIST, INGRED_LIST, STEPS_LIST, -1);
+    private static final Entity RECIPE_003 = TestUtil.createRecipeEntity(
+      ENTITY_NAME, RECIPE_NAME, "THIS IS TEST RECIPE 003 FOR THE SEARCH BY NAME TEST",
+      TAG_LIST, INGRED_LIST, STEPS_LIST, -1);
+    private static final Entity RECIPE_004 = TestUtil.createRecipeEntity(
+      ENTITY_NAME, RECIPE_NAME, "THIS IS TEST RECIPE 004 FOR THE SEARCH BY NAME TEST",
+      TAG_LIST, INGRED_LIST, STEPS_LIST, -1);
+    private static final Entity RECIPE_005 = TestUtil.createRecipeEntity(
+      ENTITY_NAME, RECIPE_NAME, "THIS IS TEST RECIPE 005 FOR THE SEARCH BY NAME TEST",
+      TAG_LIST, INGRED_LIST, STEPS_LIST, -1);
+    private static final Entity RECIPE_006 = TestUtil.createRecipeEntity(
+      ENTITY_NAME, RECIPE_NAME, "THIS IS TEST RECIPE 006 FOR THE SEARCH BY NAME TEST",
+      TAG_LIST, INGRED_LIST, STEPS_LIST, -1);
+    private static final Entity RECIPE_007 = TestUtil.createRecipeEntity(
+      ENTITY_NAME, ALT_RECIPE_NAME, "THIS IS TEST RECIPE 007 FOR THE SEARCH BY NAME TEST",
+      TAG_LIST, INGRED_LIST, STEPS_LIST, -1);
+    private static final Entity RECIPE_008 = TestUtil.createRecipeEntity(
+      ENTITY_NAME, ALT_RECIPE_NAME, "THIS IS TEST RECIPE 008 FOR THE SEARCH BY NAME TEST",
+      TAG_LIST, INGRED_LIST, STEPS_LIST, -1);
+    private static final Entity RECIPE_009 = TestUtil.createRecipeEntity(
+      ENTITY_NAME, ALT_RECIPE_NAME, "THIS IS TEST RECIPE 009 FOR THE SEARCH BY NAME TEST",
+      TAG_LIST, INGRED_LIST, STEPS_LIST, -1);
+    private static final Entity RECIPE_010 = TestUtil.createRecipeEntity(
+      ENTITY_NAME, ALT_RECIPE_NAME, "THIS IS TEST RECIPE 010 FOR THE SEARCH BY NAME TEST",
+      TAG_LIST, INGRED_LIST, STEPS_LIST, -1);
+
+    // Array of all test entities
+    private static Entity[] testRecipes = new Entity[]{RECIPE_001, RECIPE_002, RECIPE_003, RECIPE_004,
+      RECIPE_005, RECIPE_006, RECIPE_007, RECIPE_008, RECIPE_009, RECIPE_010};
```

Comment:
Seems like it would be easier to create these programmatically in the for loop below. That would also let you pass in parameters to `initializeData` to customize your test data, for example the number of recipes to create.

Reply:
Ack and fixed, though slightly different to how you proposed.